golem-worker-service-base/src/service/api_definition_service.rs

```diff
@@ -225,24 +187,22 @@ impl<Namespace: ApiNamespace, AuthCtx: Send + Sync> ApiDefinitionService<Namespa
             version: version.clone(),
         };
 
-        self.register_repo
-            .get(&key)
-            .await
-            .map(|api_definition| {
-                api_definition.map(|api_definition| ApiDefinitionAnnotated {
-                    namespace,
-                    api_definition,
-                })
-            })
-            .map_err(ApiRegistrationError::from)
+        let api_definition = self.register_repo.get(&key).await?;
+
+        let result = api_definition.map(|api_definition| ApiDefinitionAnnotated {
+            namespace,
+            api_definition,
+        });
+
+        Ok(result)
     }
 
     async fn delete(
         &self,
         api_definition_id: &ApiDefinitionId,
         version: &Version,
         auth_ctx: AuthCtx,
-    ) -> Result<ApiDefinitionIdAnnotated<Namespace>, ApiRegistrationError> {
+    ) -> Result<bool, ApiRegistrationError> {
```

Comment:
Shift this to Error, if `bool` is false. Clients don't care about this bool

Reply:
It is ok to propogate the info that "hey this api definition is deleted from this namespace", rather than just a `bool`. In other words, both ways are opinionated and can be justified, so we pick the one that will cause less  pain in the usage sites. Now if you change this to false, the cloud code base will need to change